src/btree/bt_sync.c

```diff
@@ -248,9 +266,56 @@ __wt_sync_file(WT_SESSION_IMPL *session, WT_CACHE_OP syncop)
                     btree->rec_max_txn = mod->rec_max_txn;
                 if (mod != NULL && btree->rec_max_timestamp < mod->rec_max_timestamp)
                     btree->rec_max_timestamp = mod->rec_max_timestamp;
+
+                /*
+                 * Check whether the tree is lookaside btree and verify whether the page contents
+                 * are obsolete or not by checking the visibility of the page stop time pair.
+                 */
+
+                if (is_las) {
+                    obsolete = false;
+
+                    __wt_ref_is_obsolete(session, walk, &obsolete);
+
+                    if (obsolete) {
+                        /*
+                         * Get the duplicate tree walk point before marking the current tree page as
+                         * deleted, because the duplicate tree walk expects the ref state should be
+                         * in memory.
+                         */
+                        WT_ERR(__sync_dup_walk(session, walk, flags, &prev));
+
+                        /*
+                         * Mark the page as empty and also set the parent page as dirty. This is to
+                         * ensure when the parent page is checkpointing, the empty child page will
+                         * be cleaned.
+                         */
+                        previous_state = walk->state;
+                        if (!WT_PAGE_IS_INTERNAL(walk->page)) {
+                            if (!WT_REF_CAS_STATE(session, walk, previous_state, WT_REF_LOCKED))
+                                continue;
+
+                            /* Mark the page as deleted */
+                            WT_REF_SET_STATE(walk, WT_REF_DELETED);
+                            WT_ERR(__wt_page_parent_modify_set(session, walk, true));
+                        } else {
+                            /* Mark all the child pages as obsolete, including the current page */
+                        }
+
+                        goto skip_dup_walk;
```

Comment:
I think the unconditional go to needs a comment about why we're doing it again here.

Reply:
Added a comment to explain the reason.